### PR TITLE
dev CategorySearchService does not respect ResponseGroupe

### PR DESCRIPTION
**Pull request by AliveMen**
add ReduceDetails method

### CODE REVIEW

File: VirtoCommerce.Domain/Catalog/Model/Category.cs [hunk 1]
```diff
@@ -48,7 +48,38 @@ public Category()
...
+            }
+
+            if (!categoryResponseGroup.HasFlag(CategoryResponseGroup.WithLinks))
+            {
+                Links = null;
+            }
+
+            if (!categoryResponseGroup.HasFlag(CategoryResponseGroup.WithProperties))
+            {
+                Properties = null;
```
Comment (yecli) [inline VirtoCommerce.Domain/Catalog/Model/Category.cs]: Maybe we could clear PropertyValues (need to verify if they are cleared in platform-core).

File: VirtoCommerce.Domain/Catalog/Model/CatalogProduct.cs [hunk 2]
```diff
@@ -111,5 +112,64 @@ public class CatalogProduct : AuditableEntity, ILinkSupport, ISeoSupport, IHasOu
...
+            var productResponseGroup = EnumUtility.SafeParseFlags(responseGroup, ItemResponseGroup.ItemLarge);
+
+            if (!productResponseGroup.HasFlag(ItemResponseGroup.ItemAssets))
+            {
+                Assets = null;
+            }
+
+            if (!productResponseGroup.HasFlag(ItemResponseGroup.ItemProperties))
+            {
+                Properties = null;
```
Comment (yecli) [inline VirtoCommerce.Domain/Catalog/Model/CatalogProduct.cs]: Maybe we could clear PropertyValues (need to verify if they are cleared in platform-core).

File: VirtoCommerce.Domain/Catalog/Model/Category.cs (hunk 1 above)
Comment (AliveMen) [reply VirtoCommerce.Domain/Catalog/Model/Category.cs]: it's ok with PropertyValues. 
Added missed field "Parents "

File: VirtoCommerce.Domain/Catalog/Model/CatalogProduct.cs (hunk 2 above)
Comment (AliveMen) [reply VirtoCommerce.Domain/Catalog/Model/CatalogProduct.cs]: it's ok with PropertyValues.
Added missed field

File: VirtoCommerce.Domain/Catalog/Model/CatalogProduct.cs [hunk 3]
```diff
@@ -111,5 +112,69 @@ public class CatalogProduct : AuditableEntity, ILinkSupport, ISeoSupport, IHasOu
...
+            }
+
+            if (!productResponseGroup.HasFlag(ItemResponseGroup.ItemEditorialReviews))
+            {
+                Reviews = null;
+            }
+
+            if (!productResponseGroup.HasFlag(ItemResponseGroup.ItemProperties))
+            {
+                Properties = null;
```
Comment (tatarincev) [inline VirtoCommerce.Domain/Catalog/Model/CatalogProduct.cs]: Need to set null for PropertyValues property also

File: VirtoCommerce.Domain/Catalog/Model/CatalogProduct.cs [hunk 4]
```diff
@@ -111,5 +112,69 @@ public class CatalogProduct : AuditableEntity, ILinkSupport, ISeoSupport, IHasOu
...
+            {
+                SeoInfos = null;
+            }
+
+            if (!productResponseGroup.HasFlag(ItemResponseGroup.Variations))
+            {
+                Variations = null;
+            }
+
+            if (Variations != null)
```
Comment (tatarincev) [inline VirtoCommerce.Domain/Catalog/Model/CatalogProduct.cs]: Remove this section

File: VirtoCommerce.Domain/Catalog/Model/Category.cs [hunk 5]
```diff
@@ -48,7 +48,43 @@ public Category()
...
+            }
+
+            if (!categoryResponseGroup.HasFlag(CategoryResponseGroup.WithParents))
+            {
+                Parents = null;
+            }
+
+            if (!categoryResponseGroup.HasFlag(CategoryResponseGroup.WithProperties))
+            {
+                Properties = null;
```
Comment (tatarincev) [inline VirtoCommerce.Domain/Catalog/Model/Category.cs]: Need to set null for PropertyValues property also

**Pull request merged**